File: Retos/Reto #40 - TABLA DE MULTIPLICAR [Fácil]/dart/zamudiio.dart
import 'dart:io';

void main(List<String> args) {
  print("🔢 Tabla de multiplicar 🔢");
  operation();
}

void operation() {
  print("Ingresa algun numero entero");
  String? number = stdin.readLineSync();

  if (number == null || number.isEmpty || int.tryParse(number) == null) {
    print("Porfavor ingresa un formato valido, debe ser un numero entero");
    operation();
  }

  print("La tabla de multiplicar de ${number} es la siguiente:");
  final intNumber = int.tryParse(number!)!;
  for (var i = 1; i <= 10; i++) {
    print("$i x $intNumber = ${i * intNumber}");
  }
}

File: Retos/Reto #40 - TABLA DE MULTIPLICAR [Fácil]/javascript/zamudiio.js
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});
console.log("🔢 Tabla de multiplicar 🔢");
function main() {
  rl.question("Ingresa un numero entero:", (answer) => {
      if (answer === "" || answer === null || isNaN(parseInt(answer))) {
         console.log("Porfavor ingresa un numero entero:");
         return main()
      }
      operation(parseInt(answer))
  });
}

function operation(value = 0) {
  console.log(`La tabla de multiplicacion de ${value} es:`);
  const number = value;
  for (let index = 1; index <= 10; index++) {
    console.log(`${number} x ${index} = ${index * value}`);
  }
}

main();
